### PR TITLE
[LLVM Analyzer][L1] Remove dead code

### PR DESCRIPTION
**Pull request by iarspider**
#### PR description:

Fix dead assignments [reported](https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-static-analysis/CMSSW_14_2_X_2024-09-29-0000/el8_amd64_gcc12/llvm-analysis/index.html) by LLVM Analyzer.

#### PR validation:

Bot tests

### CODE REVIEW

File: L1Trigger/DTTriggerServerPhi/src/DTTSCand.cc [hunk 1]
```diff
@@ -241,25 +241,24 @@ void DTTSCand::setBitsTsm() {
         std::cout << "DTTSCand::DTTSCand(): masking not correct: ";
         std::cout << config()->TssMasking(itk);
         std::cout << " All bits set to 1" << std::endl;
-
```
Comment (iarspider) [inline L1Trigger/DTTriggerServerPhi/src/DTTSCand.cc:244]: The code at https://github.com/cms-sw/cmssw/blob/1e22f188f24910bc4a5421ef143d256a46f9ac2c/L1Trigger/DTTriggerServerPhi/src/DTTSCand.cc#L130-L147 is similar enough to assume that the suggested changes are correct.

File: L1Trigger/HardwareValidation/interface/DEutils.h [hunk 2]
```diff
@@ -484,7 +484,7 @@ inline L1DataEmulDigi DEutils<CSCCorrelatedLCTDigiCollection_>::DEDigi(col_cit i
   dw += (itd->getBend() & ((1 << kBendBitWidth) - 1)) << (shift += kHalfstripBitWidth);
   dw += (itd->getBX() & ((1 << kBxBitWidth) - 1)) << (shift += kBendBitWidth);
   dw += (itd->getMPCLink() & ((1 << kMPCLinkBitWidth) - 1)) << (shift += kBxBitWidth);
-  dw += (itd->getCSCID() & ((1 << kCSCIdBitWidth) - 1)) << (shift += kMPCLinkBitWidth);
+  dw += (itd->getCSCID() & ((1 << kCSCIdBitWidth) - 1)) << (shift + kMPCLinkBitWidth);
```
Comment (iarspider) [inline L1Trigger/HardwareValidation/interface/DEutils.h:487]: Can revert and add `[[clang::suppress]]` if preferred

File: L1Trigger/Phase2L1GMT/interface/PreTrackMatchedMuon.h [hunk 3]
```diff
@@ -122,7 +122,7 @@ namespace Phase2L1GMT {
       bstart = wordconcat<wordtype>(w, bstart, phi_, BITSPHI);
       bstart = wordconcat<wordtype>(w, bstart, eta_, BITSETA);
       bstart = wordconcat<wordtype>(w, bstart, z0_, BITSZ0);
-      bstart = wordconcat<wordtype>(w, bstart, d0_, BITSD0);
```
Comment (iarspider) [inline L1Trigger/Phase2L1GMT/interface/PreTrackMatchedMuon.h:125]: Can revert and add `[[clang::suppress]]` if preferred

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
please test

**Comment by aloeliger** [comment]
+l1

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**